src/services/type-specifier.ts

```diff
@@ -0,0 +1,45 @@
+export type SafeFunction = (...args: unknown[]) => unknown
+
+export class TypeSpecifier {
```

Comment:
The naming scheme of the below methods, would suggest that they return boolean, due to the `isX` pattern. But in reality from my understanding of it, they return type X.

What the methods in reality are doing is checking if the given data is of a specific type, and in that case casting it to the type, and returning it.

Should the names not be something along the lines of `castToX` instead?
Or have i completely misunderstood how these methods function?

Reply:
They do indeed return a boolean, where true indicates that the value is of the given type; false meaning it does not conform the type. So using `value is type` as return type will simply augment the type knowledge that TypeScript has about the value.

Reply:
TIL. Thanks for the info.
In that case the names are just right :)